### PR TITLE
Added Stage 1 Temp Ram memory map.

### PR DESCRIPTION
**Pull request by bejeanmo**
Added a table showing the temporary memory map during Stage 1 execution.

Signed-off-by: Bejean Mosher <bejean.mosher@intel.com>

### CODE REVIEW

**Comment by bejeanmo** [comment]
I had to work out this table while working on an issue and thought it might be a good fit in the official docs.

File: source/developer-guides/memory-map.rst [hunk 1]
```diff
@@ -1,6 +1,29 @@
 Memory Map
 ------------
 
+Temporary Memory Layout
+^^^^^^^^^^^^^^^^^^^^^^^
+
+|SPN| Stage 1 temporary memory layout::
+
+          Temporary Memory
+  +------------------------------+  Top of Low MEM
```
Comment (rprangar) [inline source/developer-guides/memory-map.rst:10]: Is this labeling "Top of Low MEM" correct? Or is this 4 GB?

Comment (bejeanmo) [reply source/developer-guides/memory-map.rst:10]: You're right. This would be 4 GB. I just forgot what TOLM referred to.

**Review by rprangar — APPROVED** [review]
LGTM

**Pull request merged**